x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -191,6 +192,11 @@ private boolean assertNoPendingListeners() {
         return true;
     }
 
+    private void decrementRefCount() {
+        final boolean released = refCounter.decRef();
+        assert released == false || Files.notExists(file);
```

Comment:
Can we also assert:

```
assert released == false || evicted.get();
```
?

Reply:
Yea that sounds reasonable :) I'll open a PR

Reply:
https://github.com/elastic/elasticsearch/pull/64272